tutorials/tour/_posts/2017-02-13-classes.md

```diff
@@ -9,46 +9,103 @@ categories: tour
+```tut
+class Point(var x: Int = 0, var y: Int = 0)
 
-Note that in Scala, it isn't necessary to say `return` in order to return a value. The value returned from a method is simply the last value in the method body. In the case of the `toString` method above, the expression after the equals sign is evaluated and returned to the caller.
+val origin = new Point  // x and y are both set to 0
+val point1 = new Point(1)
+println(point1.x)  // prints 1
 
-Classes are instantiated with the `new` primitive, as follows:
+```
 
+In this version of the `Point` class, `x` and `y` have the default value `0` so no arguments are required. However, because the constructor reads arguments left to right, if you just wanted to pass in a `y` value, you would need to name the parameter.
+```
+class Point(var x: Int = 0, var y: Int = 0)
+val point2 = new Point(y=2)
+println(point2.y)  // prints 2
+```
+
+This is also a good practice to enhance clarity.
+
+## Private Members and Getter/Setter Syntax
+Members are public by default. Use the `private` access modifier
+to hide them from outside of the class.
 ```tut
-object Classes {
-  def main(args: Array[String]) {
-    val pt = new Point(1, 2)
-    println(pt)
-    pt.move(10, 10)
-    println(pt)
+class Point {
+  private var _x = 0
+  private var _y = 0
+  private val bound = 100
+
+  def x = _x
+  def x_= (newValue: Int): Unit = {
+    if (newValue < bound) _x = newValue else printWarning
+  }
+
+  def y = _y
+  def y_= (newValue: Int): Unit = {
+    if (newValue < bound) _y = newValue else printWarning
   }
+
+  private def printWarning = println("WARNING: Out of bounds")
 }
-```
 
-The program defines an executable application Classes in form of a top-level [singleton object](singleton-objects) with a `main` method. The `main` method creates a new `Point` and stores it in value `pt`. Note that values defined with the `val` construct are different from variables defined with the `var` construct (see class `Point` above) in that they do not allow updates; i.e. the value is constant.
+val point1 = new Point
+point1.x = 99
+point1.y = 101 // prints the warning
+```
+In this version of the `Point` class, the data is stored in private variables `_x` and `_y`. There are methods `def x` and `def y` for accessing the private data. `def x_=` and `def y_=` are for validating and setting the value of `_x` and `_y`. Notice the special syntax for the setters: the method has `_=` appended to the identifier of the getter and the parameters come after.
 
-Here is the output of the program:
+Primary constructor parameters with `val` and `var` are public. However, because `val`s are immutable, you can't write the following.
+```
+class Point(val x: Int, val y: Int)
+val point = new Point(1, 2)
+val.x = 3  // <-- does not compile
```

Comment:
I think you mean `point.x`, not `val.x`. 

If it's not appropriate to introduce `copy` here, maybe at least add a note that links forward to where `copy` will be introduced?